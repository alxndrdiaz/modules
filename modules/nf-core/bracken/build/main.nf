process BRACKEN_BUILD {
    tag "$meta.id"
    label 'process_medium'
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bracken:2.9--py39h1f90b4d_0':
        'biocontainers/bracken:2.9--py39h1f90b4d_0' }"
    
    input:
    tuple val(meta), path(db)

    output:
    tuple val(meta), path("$prefix"), emit: db
    path "versions.yml"             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    bracken-build \\
        -d ${db} \\
        -t $task.cpus \\
        $args
    if [[ \$(basename ${db}) != "${prefix}" ]]; then
        mv ${db}/* ${prefix}
    fi
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bracken: \$(echo \$(bracken -v) | cut -f2 -d'v')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir -p "$prefix"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bracken: \$(echo \$(bracken -v) | cut -f2 -d'v')
    END_VERSIONS
    """
}
