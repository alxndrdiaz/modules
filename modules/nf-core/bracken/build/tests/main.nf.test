nextflow_process {

    name "Test Process BRACKEN_BUILD"
    script "../main.nf"
    process "BRACKEN_BUILD"
    tag "bracken"
    tag "kraken2"
    tag "kraken2/add"
    tag "kraken2/build"
    tag "bracken/build"
    tag "gunzip"
    tag "modules"
    tag "modules_nfcore"
    
    setup {

        run("GUNZIP") {
            script "modules/nf-core/gunzip/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [],
                    file(
                        params.test_data['sarscov2']['metagenome']['prot_accession2taxid_gz'],
                        checkIfExists: true
                    )
                ])
                """
            }
        }

        run("KRAKEN2_ADD") {
            script "modules/nf-core/kraken2/add/main.nf"
            process {
                """
                input[0] = [
                    [ id:'test' ],
                        [
                        file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true),
                        file(params.test_data['sarscov2']['genome']['proteome_fasta'], checkIfExists: true)
                        ]
                    ]
                input[1] = file(params.test_data['sarscov2']['metagenome']['prot_names_dmp'], checkIfExists: true)
                input[2] = file(params.test_data['sarscov2']['metagenome']['prot_nodes_dmp'], checkIfExists: true)
                input[3] = GUNZIP.out.gunzip.map{ it[1] }
                """
            }
        }

        run("KRAKEN2_BUILD") {
            script "modules/nf-core/kraken2/build/main.nf"
            process {
                """
                input[0] = KRAKEN2_ADD.out.db
                input[1] = true
                """
            }

        }

    }

    test("sarscov2 - bracken_build") {

        when {
            process {
                """
                input[0] = KRAKEN2_BUILD.out.db
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.db.get(0).get(1) ==~ ".*/test" },
                { assert snapshot(
                        path("${process.out.db[0][1]}/database100mers.kmer_distrib")
                    ).match()
                }
            )
        }

    }
    
    test("sarscov2 - bracken_build stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = KRAKEN2_BUILD.out.db
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match()
                }
            )
        }

    }

}
