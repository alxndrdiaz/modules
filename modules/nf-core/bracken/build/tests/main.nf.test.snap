{
    "sarscov2 - bracken_build stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "db": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-17T19:44:18.560227"
    }
}